service/controller/clusterapi/v17/key/cluster.go

```diff
@@ -44,7 +44,7 @@ func ClusterCredentialSecretNamespace(cluster v1alpha1.Cluster) string {
 }
 
 func ClusterDNSZone(cluster v1alpha1.Cluster) string {
-	return clusterProviderSpec(cluster).Cluster.DNS.Domain
+	return fmt.Sprintf("%s.%s", ClusterID(&cluster), clusterProviderSpec(cluster).Cluster.DNS.Domain)
```

Comment:
Any reason `ClusterID` took a parameter as pointer type but other functions took it as value?

Reply:
It uses `LabelGetter` interface and `GetLabels()` function is defined on pointer in upstream.